hal/ia32/cpu.c

```diff
@@ -175,21 +155,21 @@ int hal_cpuReschedule(spinlock_t *spinlock, spinlock_ctx_t *scp)
 		"pushl $0;"
 		"pushl %%eax;"
 		"pushl $0;"
-		"movl %4, %%eax;"
+		"movl %1, %%eax;"
 		"call *%%eax;"
 		"cli;"
 		"addl $12, %%esp;"
 		"popl %%esp;"
```

Comment:
```suggestion
		"popl %%esp;"
```
Adding `12` to the `esp` and then loading the value from the stack to `esp`?

Reply:
We're adding 12 to `popl` 3 arguments `pushl`ed for `threads_schedule()` call. Doing `popl %%esp` and `pushl %%esp` (after `call _interrupts_multilockClear`) seems unnecessary so I've removed it (btw. it didn't hurt as `esp` is properly saved earlier with `call interrupts_pushContext`).